app/controllers/reviews_controller.rb

```diff
@@ -21,26 +21,30 @@ def edit
   # POST /reviews
   # POST /reviews.json
   def create
-    if(@restaurant)
-    @review = Review.new(review_params)
+    if (@restaurant)
+      @review = Review.new(review_params)
 
       if @review.save
-        update_rating
         render json: @review
       else
         render json: @review.errors
       end
-      end
+    else
+      render json: {:error=>"restaurant #{review_params[:restaurant_id]} dose not exist"}, status: Rack::Utils.status_code(:forbidden)
     end
+  end
 
   # PATCH/PUT /reviews/1
   # PATCH/PUT /reviews/1.json
   def update
+    if @restaurant
```

Comment:
For checking existence, use `.present?`. e.g. `@restaurant.present?`
And yes, this can safely be called on null. In Ruby, null is an object too.